### PR TITLE
EDU-223 Enable Lambda@Edge deployments

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-223

### CODE REVIEW

**Review by laurentiu-ilici — APPROVED** [review]
I'm not sure how to test this, the code looks good.

File: README.md [hunk 1]
```diff
@@ -18,7 +28,39 @@ Deployment utility for Educandu projects
...
+~~~
+
+### Arguments
+
+| Name | Description | Type | Required |
+| --- | --- | --- | --- |
+| access-key | AWS access key | string | yes |
+| secret-key | AWS secret key | string | yes |
+| lambda-env | Environment variable pair to be available to the deployed function, format `name=value` | array of string | no |
+| lambda-env-inject | Optional: Path of the JS file inside the zip file to directly inject environment variables | array of string | no |
```
Comment (adrianaluchian) [inline README.md]: Tiny thing: it's already marked as "Required: no", you could drop the "Optional: "

File: deploy-ecs.js [hunk 2]
```diff
@@ -45,17 +43,8 @@ const { ECS, Credentials } = require('aws-sdk');
...
-        const splitRegex = /^([\w]+)=(.*)$/;
-        const pairs = splitRegex.exec(nameValueString);
-
-        if (!pairs) {
-          throw new Error(`Incorrect syntax for argument 'container-env': '${nameValueString}'. Expect 'name=value'`);
-        }
-
-        return { name: pairs[1].trim(), value: pairs[2].trim() };
-      });
+    const envObj = parseEnvString((argv.containerEnv || []).join(' '));
```
Comment (adrianaluchian) [inline deploy-ecs.js:46]: Nice, simpler 👍

File: README.md (hunk 1 above)
Comment (ahelmberger) [reply README.md]: Thanks, done

File: deploy-edge.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+const REGION = 'us-east-1';
+
+export default {
+  command: 'edge',
+  describe: 'Deploy an AWS Lambda@Edge function used by a CloudFront distribution',
+  builder: yargs => yargs
+    .option('access-key', { demandOption: true, type: 'string' })
+    .option('secret-key', { demandOption: true, type: 'string' })
+    .option('lambda-env', { type: 'array', string: true })
+    .option('lambda-env-inject', { demandOption: false, type: 'string' })
```
Comment (adrianaluchian) [inline deploy-edge.js]: Tiny bit of inconsistency here with specifying `demandOption: false` or omitting it altogether.

Comment (ahelmberger) [reply deploy-edge.js]: True, I removed it here also

File: deploy-edge.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+    if (argv.wait) {
+      console.log('Waiting for distribution deployment...');
+      await cfWaitFor('distributionDeployed', {
+        Id: argv.cfDistributionId
+      });
+    }
+
+    console.log('DONE!');
+  }
```
Comment (adrianaluchian) [inline deploy-edge.js:169]: I also won't pretend I fully understand or can validate the logic here and alot is being parsed and invoked.
A small presentation of the whole integration afterwards will be useful and appreciated.

**Review by adrianaluchian — APPROVED** [review]

File: deploy-edge.js (hunk 4 above)
Comment (ahelmberger) [reply deploy-edge.js:169]: Sure

**Pull request merged**